src/main/java/org/jenkinsci/plugins/authorizeproject/AuthorizeProjectProperty.java

```diff
@@ -135,7 +138,13 @@ public Authentication authenticate(Queue.Item item) {
      */
     @Initializer(after = InitMilestone.PLUGINS_STARTED)
     public static void setStrategyCritical() {
-        Items.XSTREAM2.addCriticalField(AuthorizeProjectProperty.class, "strategy");
+        //TODO the reflection could be removed when the addCriticalField will be un-restricted (PR: https://github.com/jenkinsci/jenkins/pull/3066)
+        Method m = ReflectionUtils.findMethod(Items.XSTREAM2.getClass(), "addCriticalField", new Class[]{Class.class, String.class});
+        if(m == null) {
+            throw new RuntimeException("Class XSTREAM2 should have a method called 'addCriticalField' accepting a Class and a String as arguments");

```

Comment:
Simpler to just use `Class.getMethod` and let the initializer method throw `NoSuchMethodException`.

Reply:
done (3 exceptions in the method signature)